### PR TITLE
Kmt 206 expired licence handling

### PR DESCRIPTION
**Pull request by Thurst-ft**
Update cookie setting and getting to use FTKat as session cookie and separate FTKatL with 10min maxAge for licence list to mitigate both KMT-206 and KMT-231

Moved redirection middleware from KMT-myft to this repo to make a drier cat!
Updated the getLicenceId middleware to handle the validatedLicenceId work
Improved logging

Required as a dependency of KMT-206 on kmt-myft, kmt-overview and kmt-api

### CODE REVIEW

File: lib/helpers/isLicenceInList.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const logger = require('../logger');
+
+module.exports = function(currentLicencelist, activeLicence) {
+  if (currentLicencelist.some(licence => licence.licenceId === activeLicence)) {
+      logger.info({operation:'isLicenceInList', activeLicence:activeLicence ,IsInList: 'true'});
+    return true;
+  }
+  logger.info({operation:'isLicenceInList', activeLicence:activeLicence ,IsInList: 'false'});
```
Comment (ciprianlujeru) [inline lib/helpers/isLicenceInList.js]: We should add a `return false;` here (to keep consistency).

File: lib/initCookieSession.js [hunk 2]
```diff
@@ -1,10 +1,28 @@
 const cookieSession = require('cookie-session');
 const config = require('./config');
+//Cookie max age variant for environments 30mins prod 2 elseware
+//TODO make config var?
+//const maxAge = (config.NODE_ENV === 'production')? 30 : 2;
 
 const sessionDetails = {
-	name: 'FTkmt',
+	name: 'FTKat',
```
Comment (ciprianlujeru) [inline lib/initCookieSession.js]: You renamed this to `FTKat` here but left it as `FTkmt` here `https://github.com/Financial-Times/kmt-utilities/blob/f42a92e73d880c712cfd6a6120bd4b07dd81442b/lib/helpers/cookieHandler.js#L15`

File: lib/helpers/cookieHandler.js [hunk 3]
```diff
@@ -9,15 +10,40 @@ function decode(string) {
...
     }
 
-    logger.info({operation:'getKmtCookie', result:'getKmtCookie cookie does not exist'});
+    logger.info({operation:'getKatCookie', result:'getKatCookie cookie does not exist'});
     return;
+  },
+
+  setLicenceListCookie: function(req, res, licenceList) {
+    const cookieHandler = new Cookies(req, res, {keys:[config.get('KMT_SECRET')]});
+    cookieHandler.set('FTKatL',JSON.stringify(licenceList),{
```
Comment (ciprianlujeru) [inline lib/helpers/cookieHandler.js]: I see that you parse this below and also `decodeURIComponent`. Should we `encodeURIComponent` this `JSON.stringify`?

File: lib/middleware/redirectDefaultLicence.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+const logger = require('../logger');
+const cookieHandler = require('../helpers/cookieHandler');
+
+module.exports = function redirectDefaultLicence(req, res, next) {
```
Comment (ciprianlujeru) [inline lib/middleware/redirectDefaultLicence.js]: This seems not to be used - guessing we can remove it.

File: lib/initCookieSession.js (hunk 2 above)
Comment (dgem) [inline lib/initCookieSession.js]: @ciprianlujeru  If it's not used then I guess we should remove the method.

File: lib/helpers/cookieHandler.js (hunk 3 above)
Comment (Thurst-ft) [inline lib/helpers/cookieHandler.js]: @ciprianlujeru Fair comment. Cookie size is bigger when encode hence the reason I removed it but given this is  a bit of a hack anyway given I want to get the licence details somewhere other than a cookie I can see no harm in encoding it.

File: lib/helpers/isLicenceInList.js (hunk 1 above)
Comment (Thurst-ft) [inline lib/helpers/isLicenceInList.js]: LGTM

File: lib/middleware/redirectDefaultLicence.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+	if (undefined === req.params.licenceId && req.currentUser && req.currentUser.uuid && req.listOfLicences) {
+		const licenceId = req.listOfLicences[0].licenceId || cookieHandler.getLicenceListCookie(req, res);
```
Comment (Thurst-ft) [inline lib/middleware/redirectDefaultLicence.js]: @ciprianlujeru good spot. This is used in the default route where no licence id is provided, is it not?

**Pull request merged**